README.md

```diff
@@ -1,289 +1,610 @@
+    ```
+1.  Mount the entries of a collection into the binding `Secret` selecting the key and value from each entry
+    ```plain
+    “service.binding/endpoints”:
+      "path={.status.bootstrap},elementType=sliceOfMaps,sourceKey=type,sourceValue=url"
+    ```
+
+## Mapping Existing Values to New Values
+
+Many applications will not be able to consume the secrets exposed by Provisioned Services directly.  Teams creating Provisioned Services do not know how their services will be consumed, teams creating Applications will not know what services will be provided to them, different language families have different idioms for naming and style, and more.  Users should have a way of describing a mapping from existing values to customize the provided entries to ones that are usable directly by their applications.  This specification is described as an extension to the [Service Binding](#service-binding) specification and assumes full compatibility with it.
+
+A Service Binding Resource **MAY** define a `.spec.mappings` which is an array of `Mapping` objects.  A `Mapping` object **MUST** define `name` and `value` entries.  The value of a `Mapping` **MAY** contain zero or more tokens beginning with `((`, ending with `))`, and encapsulating a binding `Secret` key name.  The value of this `Secret` entry **MUST** be substituted into the original `value` string, replacing the token.  Once all tokens have been substituted, the new `value` **MUST** be added to the `Secret` exposed to the resource represented by `application`.
+
+### Resource Type Schema
+
+```yaml
+apiVersion: service.binding/v1alpha1
+kind: ServiceBinding
+metadata:
+  name:         # string
+spec:
+  name:         # string, optional, default: .metadata.name
+
+  application:  # PodSpec-able resource ObjectReference
+    apiVersion: # string
+    kind:       # string
+    name:       # string
+    ...
+
+  service:      # Provisioned Service-able resource ObjectReference
+    apiVersion: # string
+    kind:       # string
+    name:       # string
+    ...
+
+  mappings:     # []Mapping, optional
+  - name:       # string
+    value:      # string
+  ...
+```
 
-### Mounting and injecting binding information
+### Example Resource
 
-#### Mounting data
-Implementations of this specification **MUST** mount the binding data into the consuming application container in the following location:
+```yaml
+apiVersion: service.binding/v1alpha1
+kind: ServiceBinding
+metadata:
+  name: online-banking-to-account-service
+spec:
+  name: account-service
+  kind: database
+  provider: vmware
 
-```
-$SERVICE_BINDINGS_ROOT/<service-id>/secret/<persisted_bindings>
+  application:
+    apiVersion: apps/v1
+    kind:       Deployment
+    name:       online-banking
+
+  service:
+    apiVersion: com.example/v1alpha1
+    kind:       AccountService
+    name:       prod-account-service
+
+  mappings:
+  - name:  accountServiceUri
+    value: https://((username)):((password))@((host)):((port))/((path))
```

Comment:
Are we proposing a change in the `value` formatting here? Initially, it used to go templates format

Reply:
That is the proposal, although other extensions talk about using Go templates.  The problem with the double curly brace, is that it's valid YAML as well as being a template for Go.  Some discussion of the issue can be found in https://github.com/concourse/concourse/issues/545.

An interesting thing to note is that in another place where `{{ }}` is used, Jinja templates, the templates are [rendered first](https://stackoverflow.com/a/49031795/1384449) before being interpreted as YAML.  We don't have access to this particular strategy meaning we can't normalize the templating before it gets interpreted.

Reply:
All templating systems suffer from escaping for double templating. Helm is a common collision with `{{ }}`.

I'm not partial to any particular system so long as it's secure, but we should pick one and stick with it.

Reply:
As I see this is good for supporting a case wherein we are pulling in values from some other structure. However, referring  https://github.com/application-stacks/service-binding-specification/tree/master/sample-bindable-services/strimzi#create-a-service-binding we might have a value where we have range and if-clauses, Something like-
```
    - name: BOOTSTRAP_SERVERS
      value: |-
        {{- range .status.listeners -}}
          {{- if and (eq .type "tls") (gt (len .addresses) 0) -}}
            {{- with (index .addresses 0) -}}
              {{ .host }}:{{ .port }}
            {{- end -}}
          {{- end -}}
        {{- end -}}
```

Reply:
I don't think that there's any disagreement on support for this as part of the extension specifications and someone has already [committed to adding it](https://github.com/application-stacks/service-binding-specification/pull/48#discussion_r434712050) once this PR is merged.